Topic1/src/com/topic1/Counter.java

```diff
@@ -0,0 +1,26 @@
+package com.topic1;
+
+public class Counter {
+
+	private static Counter counter = null;
+	private int tNumber = 0;
+
+	private Counter() {
+
+	}
+
+	public static Counter getInstance() {
```

Comment:
[High] This should be a *synchronized* method.
You can check this example: https://github.com/iluwatar/java-design-patterns/blob/master/singleton/src/main/java/com/iluwatar/ThreadSafeDoubleCheckLocking.java

or read the *Thread Safe Singleton* in this post: http://www.journaldev.com/1377/java-singleton-design-pattern-best-practices-with-examples.

Also read the 4 steps for a singleton here: http://www.javabeginner.com/learn-java/java-singleton-design-pattern. It's important to override *clone* method for singletons